File: 150.java
// Leetcode 150 Evaluate Reverse olish Notation
// Reference: https://leetcode.com/problems/evaluate-reverse-polish-notation/solutions/47430/java-accepted-code-stack-implementation/?envType=featured-list&envId=top-interview-questions?envType=featured-list&envId=top-interview-questions

import java.util.Stack;

class Solution {
    public int evalRPN(String[] tokens) {
        int a, b;
        Stack<Integer> S = new Stack<Integer>();
        for (String s : tokens){
            if (s.equals("+")){
                
                S.add(S.pop()+S.pop());

            } else if(s.equals("/")) {
				b = S.pop();
				a = S.pop();
				S.add(a / b);
            
            } else if(s.equals("*")) {
				S.add(S.pop() * S.pop());
			
            } else if(s.equals("-")) {
				b = S.pop();
				a = S.pop();
				S.add(a - b);

			} else {
                S.add(Integer.parseInt(s));
            }
        }
        return S.pop();

        
    }
}

File: 150.js
// Leetcode 150 Evaluate Reverse olish Notation
// Reference: https://leetcode.com/problems/evaluate-reverse-polish-notation/solutions/486566/javascript-stack-solution/?envType=featured-list&envId=top-interview-questions?envType=featured-list&envId=top-interview-questions

/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack = [];
    let ops = {
        '+': (a, b) => a + b,
        '-': (a, b) => a - b,
        '*': (a, b) => a * b,
        '/': (a, b) => a / b >= 0 ? Math.floor(a/b): Math.ceil(a/b)
    };

    for (let t of tokens){
        if (ops[t]){
            let top = stack.pop();
            let second = stack.pop();
            stack.push(ops[t](second, top));
        } else {
            stack.push(Number(t));
        }
    }
    return stack.pop();
    
};